src/hooks/useGraphSyncStatus.ts

```diff
@@ -12,91 +14,90 @@ const ethGraphQuery =
+}
+
 async function fetchGraph(url: string, queryBody: string): Promise<Response> {
   try {
     const response = await fetch(url, {
       method: 'POST',
       body: queryBody
     })
 
-    return response.json()
+    return await response.json()
   } catch (error) {
     console.error('Error parsing json: ' + error.message)
   }
 }
 
-export function useGraphSyncStatus(): {
-  isGraphSynced: boolean
-  blockNumber: number
-  graphBlockNumber: number
-  state: 'success' | 'warning'
-} {
-  const { config } = useOcean()
-  const { web3 } = useWeb3()
-  const [state, setState] = useState<'success' | 'warning'>()
-  const [blockNumber, setBlockNumber] = useState<number>()
-  const [graphBlockNumber, setGraphBlockNumber] = useState<number>()
-  const [isGraphSynced, setIsGraphSynced] = useState(true)
-
-  async function getBlockFromPolygon() {
-    console.log('Polygon')
-    const web3Polygon = new Web3(config.nodeUri)
-    const blockNumber = await web3Polygon.eth.getBlockNumber()
-    setBlockNumber(blockNumber)
+async function getBlockHead(config: ConfigHelperConfig) {
+  // for ETH main, get block from graph fetch
+  if (config.network === 'mainnet') {
+    const response: any = await fetchGraph(ethGraphUrl, ethGraphQuery)
+    return Number(response.data.blocks[0].number)
   }
 
-  async function getBlockFromWeb3() {
-    console.log('withProvider')
-    const blockNumber = await web3.eth.getBlockNumber()
-    setBlockNumber(blockNumber)
-  }
+  // for everything else, create new web3 instance with infura
+  const web3 = new Web3(config.nodeUri)
+  const blockHead = await web3.eth.getBlockNumber()
+  return blockHead
+}
 
-  async function getBlockFromETH() {
-    const response: any = await fetchGraph(ethGraphUrl, ethGraphQuery)
-    setBlockNumber(response.data.blocks[0].number)
-  }
+async function getBlockSubgraph(subgraphUri: string) {
+  const response: any = await fetchGraph(
+    `${subgraphUri}/subgraphs/name/oceanprotocol/ocean-subgraph`,
+    graphQuery
+  )
+  const blockNumberGraph = Number(response.data._meta.block.number)
+  return blockNumberGraph
+}
 
-  async function getBlockFromGraph() {
-    const response: any = await fetchGraph(
-      `${config.subgraphUri}/subgraphs/name/oceanprotocol/ocean-subgraph`,
-      graphQuery
-    )
-    setGraphBlockNumber(response.data._meta.block.number)
-  }
+export function useGraphSyncStatus(): UseGraphSyncStatus {
+  const { config } = useOcean()
+  const { block } = useWeb3()
+  const [blockGraph, setBlockGraph] = useState<number>()
+  const [blockHead, setBlockHead] = useState<number>()
+  const [isGraphSynced, setIsGraphSynced] = useState(true)
 
+  // Get and set head block
   useEffect(() => {
-    if (!config) return
+    if (!config || !config.nodeUri) return
 
-    getBlockFromGraph()
-    const timer = setInterval(() => getBlockFromGraph(), refreshInterval)
+    async function initBlockHead() {
+      const blockHead = block || (await getBlockHead(config))
```

Comment:
I think we should only use the `await getBlockHead(config)` to get the block number. I just tested and the block number is not changing inside the `Web3` provider when I change the network. So I get stocked with the initial block number

Reply:
sorry, just pushed some fixed, can you try again? The block retrieved via `web3` should always be the source of truth, hence I put that into our `Web3Provider`. So if wallets are connected this will always kick in. So the `await getBlockHead(config)` then is only the fallback. Cause right now, `getBlockHead()` only works for ETH mainnet, because `blocklytics/ethereum-blocks` is only ETH mainnet. So we would need to then start finding ways to get this for all ETH networks, which becomes quite cumbersome.

The ideal case is that we always get the block from web3, either with wallet, or infura-like service.

Reply:
I get it. Tested it and works with `networkId` dependency added to the hook, now it's changing